patches/minecraft/net/minecraft/item/ItemStack.java.patch

```diff
@@ -214,16 +214,20 @@
        return list;
     }
  
-@@ -817,7 +841,7 @@
+@@ -817,9 +841,10 @@
              }
           }
        } else {
 -         multimap = this.func_77973_b().func_111205_h(p_111283_1_);
 +         multimap = this.func_77973_b().getAttributeModifiers(p_111283_1_, this);
        }
- 
+-
++      multimap = com.google.common.collect.ArrayListMultimap.create(multimap);
++      net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.ItemModifierEvent(this, p_111283_1_, multimap));
```

Comment:
Part of me wonders if there is some good way that we could maybe "cache" the result of this as of 1.16 Vanilla caches in an ImmutableMultiMap the attributes items themselves return. Alternatively should this at least be returning an immutable map from this method in ItemStack after making it mutable for purposes of firing the event so that people don't accidentally "inject" attributes outside of using the event onto item stacks.

Reply:
I don't think this can be reliably cached as the initial map is not deterministic. Making it immutable again is fine however it would add more overhead.

Reply:
Ya I wasn't sure if there is a good way to cache it which is why I was mentioning wondering. I do think  having it be immutable would be better though, or at least unnmodifiable given nothing would have access to the local ArrayListMultimap to accidentally mutate it. Though I am not sure if there is a unnmodifiable view for multimaps.